agent/provider/src/hardware.rs

```diff
@@ -65,24 +65,45 @@ pub struct Resources {
 }
 
 impl Resources {
-    pub fn new_empty() -> Self {
+    pub fn try_with_config(work_dir: &Path, config: &ProviderConfig) -> Result<Self, Error> {
+        if config.rt_cores.is_some() || config.rt_mem.is_some() || config.rt_storage.is_some() {
+            let max_caps = Self::max_caps(work_dir)?;
```

Comment:
Use `default_caps` here. It might be surprising that for example if you specify storage in config, it takes not 0.7 but max of memory (which is unrelated to storage).

Reply:
These caps serve as an upper limit for the currently active hardware profile.

On a second thought, I think that `default_caps` should only be used while creating a default profile, otherwise the user won't be able to go beyond the upper resource limit.